### PR TITLE
Remove `optuna dashboard`

### PR DESCRIPTION
**Pull request by himkt**
~This PR updates the deprecation period of Optuna built-in dashboard command. `_Dashboard.take_action` is annotated to be removed in v3.0.0 but we're now on the migration to [`optuna-dashbaord`](https://github.com/optuna/optuna-dashboard). So, in this PR, I update `removed_version` to `v4.0.0`.~

~🔗  https://github.com/optuna/optuna/pull/2559#pullrequestreview-626783569~
~🔗  https://github.com/optuna/optuna/issues/2902~

In this PR, I remove `dashboard` subcommand based on its deprecation period.

### CODE REVIEW

**Review by not522 — APPROVED** [review]
In #2559, we choosed `v3.0.0` as `removed_version`, because this feature is experimental and we may use `optuna dashboard` as alias of `optuna-dashboard`. But this alias will cause circular dependency and we do not introduce it.
The default `removed_version` is the next next major version after the deprecated version, i.e., `v4.0.0`. I agree with this PR.
@g-votte, Do you have any comments on this change?

**Comment by himkt** [comment]
I discussed with @g-votte and @not522 and decided that we remove `dashbaord` command. I revise this PR.

**Review by not522 — APPROVED** [review]
LGTM!

File: setup.py [hunk 1]
```diff
@@ -86,7 +86,6 @@ def get_extras_require() -> Dict[str, List[str]]:
         "testing": [
             # TODO(toshihikoyanase): Remove the version constraint after resolving the issue
             # https://github.com/optuna/optuna/issues/1000.
```
Comment (nzw0301) [inline setup.py:88]: Can we remove these lines since it issue refers to `bokeh` that will be removed in this PR?

Comment (himkt) [reply setup.py:88]: Thank you. I removed in 80df813.

Comment (nzw0301) [reply setup.py:88]: still in https://github.com/optuna/optuna/blob/80df8136168e283a7dab7d93ac6f5ad51bc2ded5/setup.py#L87-L88

Comment (himkt) [reply setup.py]: :bow: 59d11ea

**Comment by g-votte** [comment]
`bokeh` still remains here:
https://github.com/optuna/optuna/blob/master/optuna/__init__.py#L64

We may want to replace this with another optional dependency.

**Comment by himkt** [comment]
@g-votte 

> We may want to replace this with another optional dependency.

`_LazyImport`  is not used only for `optuna.type_checking` and we don't have example of `_lazyImport` that imports third-party libraries. So I simply removed e.g. from the comment (f076e86). What do you think about it?

**Comment by g-votte** [comment]
@himkt 
Thanks for your quick action. I agree that we can simply get rid of the specific library name from the comment.

Just FYI, lazy import is going to be applied again for third party modules with this PR:
https://github.com/optuna/optuna/pull/3000

**Review by g-votte — APPROVED** [review]
LGTM

@nzw0301 
Please merge if you don't have any other concern.

**Comment by nzw0301** [comment]
Thanks! I confirmed that `bokeh` and unrelated `dashboard` are not in this repo anymore.

**Pull request merged**